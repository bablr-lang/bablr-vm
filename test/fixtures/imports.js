import     def,{   imported  as local, shorthand, name as name} from  'bar';
