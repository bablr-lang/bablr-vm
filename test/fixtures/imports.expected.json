{
  "type": "Program",
  "sourceType": "module",
  "body": [
    {
      "type": "ImportDeclaration",
      "specifiers": [
        {
          "type": "ImportDefaultSpecifier",
          "local": {
            "type": "Identifier",
            "name": "def",
            "cstTokens": [
              {
                "type": "Identifier",
                "value": "def"
              }
            ]
          },
          "cstTokens": [
            {
              "type": "Reference",
              "value": "local"
            }
          ]
        },
        {
          "type": "ImportSpecifier",
          "local": {
            "type": "Identifier",
            "name": "local",
            "cstTokens": [
              {
                "type": "Identifier",
                "value": "local"
              }
            ]
          },
          "imported": {
            "type": "Identifier",
            "name": "imported",
            "cstTokens": [
              {
                "type": "Identifier",
                "value": "imported"
              }
            ]
          },
          "cstTokens": [
            {
              "type": "Reference",
              "value": "imported"
            },
            {
              "type": "Whitespace",
              "value": "  "
            },
            {
              "type": "Identifier",
              "value": "as"
            },
            {
              "type": "Whitespace",
              "value": " "
            },
            {
              "type": "Reference",
              "value": "local"
            }
          ]
        },
        {
          "type": "ImportSpecifier",
          "local": {
            "type": "Identifier",
            "name": "shorthand",
            "cstTokens": [
              {
                "type": "Identifier",
                "value": "shorthand"
              }
            ]
          },
          "imported": {
            "type": "Identifier",
            "name": "shorthand",
            "cstTokens": [
              {
                "type": "Identifier",
                "value": "shorthand"
              }
            ]
          },
          "cstTokens": [
            {
              "type": "Reference",
              "value": "imported"
            }
          ]
        },
        {
          "type": "ImportSpecifier",
          "local": {
            "type": "Identifier",
            "name": "name",
            "cstTokens": [
              {
                "type": "Identifier",
                "value": "name"
              }
            ]
          },
          "imported": {
            "type": "Identifier",
            "name": "name",
            "cstTokens": [
              {
                "type": "Identifier",
                "value": "name"
              }
            ]
          },
          "cstTokens": [
            {
              "type": "Reference",
              "value": "imported"
            },
            {
              "type": "Whitespace",
              "value": " "
            },
            {
              "type": "Identifier",
              "value": "as"
            },
            {
              "type": "Whitespace",
              "value": " "
            },
            {
              "type": "Reference",
              "value": "local"
            }
          ]
        }
      ],
      "source": {
        "type": "Literal",
        "value": "bar",
        "cstTokens": [
          {
            "type": "LeftPunctuator",
            "value": "'"
          },
          {
            "type": "Text",
            "value": "bar"
          },
          {
            "type": "RightPunctuator",
            "value": "'"
          }
        ]
      },
      "cstTokens": [
        {
          "type": "Keyword",
          "value": "import"
        },
        {
          "type": "Whitespace",
          "value": "     "
        },
        {
          "type": "Reference",
          "value": "specifiers"
        },
        {
          "type": "Punctuator",
          "value": ","
        },
        {
          "type": "LeftPunctuator",
          "value": "{"
        },
        {
          "type": "Whitespace",
          "value": "   "
        },
        {
          "type": "Reference",
          "value": "specifiers"
        },
        {
          "type": "Punctuator",
          "value": ","
        },
        {
          "type": "Whitespace",
          "value": " "
        },
        {
          "type": "Reference",
          "value": "specifiers"
        },
        {
          "type": "Punctuator",
          "value": ","
        },
        {
          "type": "Whitespace",
          "value": " "
        },
        {
          "type": "Reference",
          "value": "specifiers"
        },
        {
          "type": "RightPunctuator",
          "value": "}"
        },
        {
          "type": "Whitespace",
          "value": " "
        },
        {
          "type": "Keyword",
          "value": "from"
        },
        {
          "type": "Whitespace",
          "value": "  "
        },
        {
          "type": "Reference",
          "value": "source"
        },
        {
          "type": "Punctuator",
          "value": ";"
        }
      ]
    }
  ],
  "cstTokens": [
    {
      "type": "Reference",
      "value": "body"
    },
    {
      "type": "Whitespace",
      "value": "\n"
    }
  ]
}
